test_flake8.py

```diff
@@ -152,3 +152,12 @@ def test_strict(testdir):
     testdir.makepyfile("")
     result = testdir.runpytest("--strict", "--flake8")
     assert result.ret == 0
+
+
+def test_junit_classname(testdir):
+    testdir.makepyfile("")
+    result = testdir.runpytest("--flake8", "--junit-xml=TEST.xml")
+    junit = testdir.tmpdir.join("TEST.xml")
+    with open(str(junit)) as j_file:
+        j_text = j_file.read()
+    assert result.ret == 0 and 'classname=""' not in j_text
```

Comment:
It might be better to split it into 2 asserts: `assert  result.ret == 0 ` and `assert 'classname=""' not in j_text`